jlh/WeatherApp/app/src/main/java/com/e_conomic/weatherapp/domain/mappers/ForecastDataMapper.kt

```diff
@@ -0,0 +1,33 @@
+package com.e_conomic.weatherapp.domain.mappers
+
+import com.e_conomic.weatherapp.data.Forecast
+import com.e_conomic.weatherapp.data.ForecastResult
+import com.e_conomic.weatherapp.domain.model.ForecastList
+import java.text.DateFormat
+import java.util.*
+import com.e_conomic.weatherapp.domain.model.Forecast as ModelForecast
+
+class ForecastDataMapper {
+
+    fun convertFromDataModel(forecast: ForecastResult): ForecastList {
+        return ForecastList(forecast.city.name, forecast.city.country,
+                convertForecastListToDomain(forecast.list))
+    }
+
+    private fun convertForecastListToDomain(forecastList: List<Forecast>): List<ModelForecast> {
+        return forecastList.map { convertForecastItemToDomain(it) }
```

Comment:
We often write out the parameter explicitly as it makes it obvious what it is. Here (also renaming `convertForecastItemToDomain` ->  `convertToModelForecast`) it could be 
```
return forecastList.map { forecast -> convertToModelForecast(forecast) }
```